### ISSUE TITLE
In fastapi==0.89.0 add_api_route fails with FileResponse

### ISSUE THREAD

**Issue by DannyVarod** [tiangolo/fastapi#5857]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.responses import FileResponse

from pathlib2 import Path

app = FastAPI()

@app.get("/favicon.ico")
def favicon() -> FileResponse:
    result = FileResponse(
        path='C:\\TEMP\\favicon.ico', -- any valid path, this is an example for Windows
        status_code=200,
        media_type='image/png',
        filename='favicon.ico',
    )
    return result
```

### Description

When using FileResponse as documented here: https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse
and upgrading to fastapi==0.89.0 (works on 0.88.0) exception is thrown:

   File "/usr/local/lib/python3.7/site-packages/fastapi/routing.py", line 658, in decorator
     generate_unique_id_function=generate_unique_id_function,
   File "/usr/local/lib/python3.7/site-packages/fastapi/routing.py", line 598, in add_api_route
     generate_unique_id_function=current_generate_unique_id,
   File "/usr/local/lib/python3.7/site-packages/fastapi/routing.py", line 401, in __init__
     name=response_name, type_=self.response_model
   File "/usr/local/lib/python3.7/site-packages/fastapi/utils.py", line 92, in create_response_field
     ) from None
 fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'starlette.responses.FileResponse'> is a valid pydantic field type

### Operating System

Linux, Windows

### Operating System Details

Checked on GitHub actions (Linux) with various FastAPI versions and Python 3.7.
With versions until 0.88.0 (including) this works. With 0.89.0 this fails.
Locally using Python 3.7 above Windows 10 with various FastAPI versions (including 0.88.0, 0.89.0).

### FastAPI Version

0.89.0

### Python Version

3.7

**Comment by yinziyan1206** [comment]
It's a bug. The `response_model` should be a pydantic model or basic type.

**Comment by anthonycorletti** [comment]
Seeing similar behavior with `RedirectResponse` :( 

```
# main.py
from fastapi import FastAPI
from fastapi.responses import RedirectResponse

app = FastAPI()

@app.get("/fastapi", response_class=RedirectResponse)
async def redirect_fastapi() -> RedirectResponse:
    return RedirectResponse("https://fastapi.tiangolo.com")
```

```
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'starlette.responses.RedirectResponse'> is a valid pydantic field type
```

**Comment by PeterBaker0** [comment]
We are experiencing the same issue in our code base after updates. Thanks.

**Comment by phillipuniverse** [comment]
Same issue with `JsonResponse` and `NoReturn`:

```python
from Typing import NoReturn

@app.post("/exception")
async def ok(body: dict[str, str]) -> NoReturn:
    raise ValueError("Failed!")
```

**Comment by zackees** [comment]
Also hitting this bug with RedirectResponse:

```
ersonalassistantmonitor-collector-app-1  | Traceback (most recent call last):
personalassistantmonitor-collector-app-1  |   File "/usr/local/bin/uvicorn", line 8, in <module>
personalassistantmonitor-collector-app-1  |     sys.exit(main())
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
personalassistantmonitor-collector-app-1  |     return self.main(*args, **kwargs)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
personalassistantmonitor-collector-app-1  |     rv = self.invoke(ctx)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
personalassistantmonitor-collector-app-1  |     return ctx.invoke(self.callback, **ctx.params)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
personalassistantmonitor-collector-app-1  |     return __callback(*args, **kwargs)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 404, in main
personalassistantmonitor-collector-app-1  |     run(
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 569, in run
personalassistantmonitor-collector-app-1  |     server.run()
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 60, in run
personalassistantmonitor-collector-app-1  |     return asyncio.run(self.serve(sockets=sockets))
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
personalassistantmonitor-collector-app-1  |     return loop.run_until_complete(main)
personalassistantmonitor-collector-app-1  |   File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 67, in serve
personalassistantmonitor-collector-app-1  |     config.load()
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 477, in load
personalassistantmonitor-collector-app-1  |     self.loaded_app = import_from_string(self.app)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
personalassistantmonitor-collector-app-1  |     module = importlib.import_module(module_str)
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
personalassistantmonitor-collector-app-1  |     return _bootstrap._gcd_import(name[level:], package, level)
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
personalassistantmonitor-collector-app-1  |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
personalassistantmonitor-collector-app-1  |   File "/app/./personalmonitor_collector/app.py", line 105, in <module>
personalassistantmonitor-collector-app-1  |     async def redirect_default() -> RedirectResponse:
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 633, in decorator
personalassistantmonitor-collector-app-1  |     self.add_api_route(
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 572, in add_api_route
personalassistantmonitor-collector-app-1  |     route = route_class(
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 400, in __init__
personalassistantmonitor-collector-app-1  |     self.response_field = create_response_field(
personalassistantmonitor-collector-app-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/utils.py", line 90, in create_response_field
personalassistantmonitor-collector-app-1  |     raise fastapi.exceptions.FastAPIError(
personalassistantmonitor-collector-app-1  | fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'starlette.responses.RedirectResponse'> is a valid pydantic 
field type
```

**Comment by jvbmov** [comment]
Idem for

`from fastapi.responses import StreamingResponse`

Log results in:

` File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 400, in __init__
    self.response_field = create_response_field(
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/utils.py", line 90, in create_response_field
    raise fastapi.exceptions.FastAPIError(
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'starlette.responses.StreamingResponse'> is a valid pydantic field type`

**Comment by rra** [comment]
This looks like the same problem as https://github.com/tiangolo/fastapi/issues/5861. Returning a `Response` or a subclass of it (as documented at https://fastapi.tiangolo.com/advanced/response-directly/) seems to have accidentally broken in the latest release.

**Comment by zackees** [comment]
I can only imagine how many people this is breaking right now.

@tiangolo needs to pull the 0.89 release from pypi

**Comment by phillipuniverse** [comment]
Confirmed fixed in 0.89.1 via #5855

**Comment by tiangolo** [comment]
Thanks for the report and discussion everyone! :coffee: 

The simple use case of a single `Response` class is now supported in FastAPI 0.89.1 (just release) :tada:

And for the other use cases where you want to have more complex type annotations (e.g. unions, other types), you can use `response_model=None`.

There's a lot of new docs about it here: https://fastapi.tiangolo.com/tutorial/response-model/#other-return-type-annotations
